### PR TITLE
Remove AABB implementation from the API

### PR DESCRIPTION
**Pull request by kashike**
https://github.com/SpongePowered/SpongeAPI/pull/2269

### CODE REVIEW

File: src/mixins/java/org/spongepowered/common/mixin/core/world/server/ServerBossInfoMixin.java [hunk 1]
```diff
@@ -110,9 +110,9 @@ public void bossBarNameChanged(final BossBar bar, final Component oldName, final
     }
 
     @Override
-    public void bossBarPercentChanged(final BossBar bar, final float oldPercent, final float newPercent) {
-        if (Math.abs(newPercent - this.impl$lastSentPercent) > EPSILON) {
-            this.impl$lastSentPercent = newPercent;
+    public void bossBarProgressChanged(final BossBar bar, final float oldProgress, final float newProgress) {
```
Comment (Zidane) [inline src/mixins/java/org/spongepowered/common/mixin/core/world/server/ServerBossInfoMixin.java]: impl$

File: src/test/java/org/spongepowered/common/util/AABBImplTest.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+import java.util.Random;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.spongepowered.api.util.AABB;
+import org.spongepowered.api.util.Tuple;
+import org.spongepowered.math.vector.Vector3d;
+import org.spongepowered.math.vector.Vector3i;
+
+public class AABBImplTest {
```
Comment (Zidane) [inline src/test/java/org/spongepowered/common/util/AABBImplTest.java]: Hmmm

Comment (Faithcaio) [reply src/test/java/org/spongepowered/common/util/AABBImplTest.java]: good we **finally** moved this to impl

File: src/main/java/org/spongepowered/common/util/AABBImpl.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+package org.spongepowered.common.util;
+
+import java.util.Objects;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.spongepowered.api.util.AABB;
+import org.spongepowered.api.util.Tuple;
+import org.spongepowered.math.vector.Vector3d;
+
+public final class AABBImpl implements AABB {
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/util/AABBImpl.java:34]: I know this drives you nuts but doesn't follow the rest of Sponge, should be SpongeAABB.

File: src/test/java/org/spongepowered/common/util/AABBImplTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+import java.util.Random;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.spongepowered.api.util.AABB;
+import org.spongepowered.api.util.Tuple;
+import org.spongepowered.math.vector.Vector3d;
+import org.spongepowered.math.vector.Vector3i;
+
+final class AABBImplTest {
```
Comment (Zidane) [inline src/test/java/org/spongepowered/common/util/AABBImplTest.java:35]: SpongeAABBTest

**Comment by Zidane** [comment]
Manually merged, thanks kashike.

**Pull request closed without merging**